x/slashing/keeper/infractions.go

```diff
@@ -142,3 +151,29 @@ func (k Keeper) SlashJailAndUpdateSigningInfo(ctx sdk.Context, consAddr sdk.Cons
 	)
 	return signInfo
 }
+
+func (k Keeper) ResizeMissedBlockArray(missedInfo types.ValidatorMissedBlockArray, signInfo types.ValidatorSigningInfo, window int64, index int64) (types.ValidatorMissedBlockArray, types.ValidatorSigningInfo, int64) {
+	// we need to resize the missed block array AND update the signing info accordingly
+	switch {
+	case missedInfo.WindowSize < window:
+		// missed block array too short, lets expand it
+		boolArray := k.ParseBitGroupsToBoolArray(missedInfo.MissedBlocks, missedInfo.WindowSize)
+		newArray := make([]bool, window)
+		copy(newArray[0:index], boolArray[0:index])
+		if index+1 < missedInfo.WindowSize {
+			// insert `0`s corresponding to the difference between the new window size and old window size
+			copy(newArray[index+(window-missedInfo.WindowSize):], boolArray[index:])
```

Comment:
would it just be easier to have newArray filled with `0`'s and then update [0:index] in both? I guess i'm confused what the last line here is doing

Reply:
we need to update 0:index and then insert N corresponding to the number of blocks added to the signing window, and THEN include the rest of the array. This is because we want to update the X oldest value (x=signwindow).

Example
```
[0, 1, 1, 0] index = 2 (resize from 4 -> 6) 
because the above index = 2 is the 4th oldest value (we know this because it was about to be updated), it should only be updated after updating the 6th and 5th oldest values.
therefore we result with the expanded array

[0, 1, 0, 0, 1, 0]
in this, index 2 and 3 represent the expanded values and we update index = 2 because that represents the 6th oldest block
```

Lmk if this doesn't make sense or if i'm mistunderstanding your suggested modification